src/griffe/dataclasses.py

```diff
@@ -1300,7 +1300,12 @@ def as_dict(self, **kwargs: Any) -> dict[str, Any]:
             A dictionary.
         """
         base = super().as_dict(**kwargs)
-        base["filepath"] = str(self._filepath) if self._filepath else None
+
+        filepath = None
+        if self._filepath:
+            filepath = [str(p) for p in self._filepath] if isinstance(self._filepath, list) else str(self.filepath)
+
+        base["filepath"] = filepath
```

Comment:
Yep, good catch!

I'd rather assign directly into the dict item though:

```suggestion
        if isinstance(self._filepath, list):
            base["filepath"] = [str(path) for path in self._filepath]
        elif self._filepath:
            base["filepath"] = str(self._filepath)
        else:
            base["filepath"] = None
```

Reply:
I initially went with that but thought it might have been a bit too much code duplication, but works for me too!